Mage.Sets/src/mage/cards/l/LeoriSparktouchedHunter.java

```diff
@@ -0,0 +1,162 @@
+import mage.abilities.keyword.FlyingAbility;
+import mage.abilities.keyword.VigilanceAbility;
+import mage.abilities.mana.ActivatedManaAbilityImpl;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.choices.Choice;
+import mage.choices.ChoicePlaneswalkerType;
+import mage.constants.*;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.game.stack.StackAbility;
+import mage.players.Player;
+
+import java.util.UUID;
+
+/**
+ * @author Susucr
+ */
+public final class LeoriSparktouchedHunter extends CardImpl {
+
+    public LeoriSparktouchedHunter(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{U}{R}{W}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.ELEMENTAL);
+        this.subtype.add(SubType.CAT);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Vigilance
+        this.addAbility(VigilanceAbility.getInstance());
+
+        // Whenever Leori, Sparktouched Hunter deals combat damage to a player, choose a planeswalker type. Until end of turn, whenever you activate an ability of a planeswalker of that type, copy that ability. You may choose new targets for the copies.
+        this.addAbility(new DealsCombatDamageToAPlayerTriggeredAbility(
+            new LeoriSparktouchedHunterEffect(),
+            false
+        ));
+    }
+
+    private LeoriSparktouchedHunter(final LeoriSparktouchedHunter card) {
+        super(card);
+    }
+
+    @Override
+    public LeoriSparktouchedHunter copy() {
+        return new LeoriSparktouchedHunter(this);
+    }
+}
+
+
+class LeoriSparktouchedHunterEffect extends OneShotEffect {
+
+    LeoriSparktouchedHunterEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "choose a planeswalker type. Until end of turn, whenever you activate an ability " +
+            "of a planeswalker of that type, copy that ability. You may choose new targets for the copies.";
+    }
+
+    private LeoriSparktouchedHunterEffect(final LeoriSparktouchedHunterEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public LeoriSparktouchedHunterEffect copy() {
+        return new LeoriSparktouchedHunterEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        Choice choice = new ChoicePlaneswalkerType(game.getObject(source));
+        if (!controller.choose(outcome, choice, game)) {
+            return false;
+        }
+
+        SubType subType = SubType.fromString(choice.getChoice());
+        if (subType == null) {
+            return false;
+        }
+
+        if (controller != null) {
```

Comment:
this was already null checked right?

Reply:
Indeed, that one check is not required.
I might be developping some paranoia on those null checks.